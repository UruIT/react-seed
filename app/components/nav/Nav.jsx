import React from 'react';
import { NavLink } from 'react-router-dom';
import { nav, active } from './Nav.scss';

const Nav = () => (
	<ul className={nav}>
		<li>
			<NavLink to="/" activeClassName={active} exact>
				Home
			</NavLink>
		</li>
		<li>
			<NavLink to="/about" activeClassName={active} exact>
				About
			</NavLink>
		</li>
	</ul>
);

export default Nav;
