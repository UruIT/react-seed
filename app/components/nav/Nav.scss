ul.nav {
	border: 1px solid;
	border-width: 1px 0;
	list-style: none;
	text-align: center;

	li {
		display: inline;
	}

	a {
		display: inline-block;
		padding: 10px;
	}
}

.active {
	font-weight: 600;
}
