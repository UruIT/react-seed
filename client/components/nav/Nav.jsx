import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Nav.scss';
import { FormattedMessage } from 'react-intl';

let { nav, active } = styles;

const Nav = () =>
	<ul className={nav}>
		<li>
			<NavLink to="/" activeClassName={active} exact>
				<FormattedMessage id="home" defaultMessage="Home" />
			</NavLink>
		</li>
		<li>
			<NavLink to="/about" activeClassName={active} exact>
				<FormattedMessage id="about" defaultMessage="About" />
			</NavLink>
		</li>
	</ul>;

export default Nav;
