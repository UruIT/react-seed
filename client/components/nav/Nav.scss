@import '~styles/_palette.scss';
@import '~styles/_mixins.scss';

ul.nav {
	width: 150px;
	margin: 0 auto;
	border-bottom: 1px dashed $black-2;
	border-width: 1px 0;
	list-style: none;
	text-align: center;
	padding: 0;

	li {
		display: inline-block;

		&:not(:last-child)::after {
			content: '|';
			color: $black-2;
		}
	}

	.link {
		display: inline-block;
		text-decoration: none;
		padding: 8px;
		margin: 0 8px;
		@include blurry();
	}

	.active {
		font-weight: 600;
	}
}
