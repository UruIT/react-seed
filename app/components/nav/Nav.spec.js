import React from 'react';
import Nav from './Nav';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import withContext from '../__mocks__/context-hoc';

test('<Nav/>', () => {
	const ROUTER = {
		router: {
			route: {
				location: jest.fn()
			},
			history: {
				createHref: jest.fn(),
				replace: jest.fn(),
				push: jest.fn()
			}
		}
	};
	const NavWithContext = withContext(ROUTER)(Nav);
	const tree = renderer.create(<NavWithContext />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('shallow <Nav/>', () => {
	const renderer = new ShallowRenderer();
	const tree = renderer.render(<Nav />);

	expect(tree).toMatchSnapshot();
});
