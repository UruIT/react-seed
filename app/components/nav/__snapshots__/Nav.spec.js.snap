// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> 1`] = `
<ul
  className="nav"
>
  <li>
    <a
      className={undefined}
      href={undefined}
      onClick={[Function]}
      style={undefined}
    >
      Home
    </a>
  </li>
  <li>
    <a
      className={undefined}
      href={undefined}
      onClick={[Function]}
      style={undefined}
    >
      About
    </a>
  </li>
</ul>
`;

exports[`shallow <Nav/> 1`] = `
<ul
  className="nav"
>
  <li>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/about"
    >
      About
    </NavLink>
  </li>
</ul>
`;
